### PR TITLE
SB-22584 Adding Entry and Exit Log, traceId

### PR DESCRIPTION
**Pull request by sknirmalkar89**

### CODE REVIEW

File: group-actors/src/main/java/org/sunbird/actors/BaseActor.java [hunk 1]
```diff
@@ -63,10 +63,10 @@ public void onReceive(Object message) throws Throwable {
    */
   protected void onReceiveException(String callerName, Exception exception) throws Exception {
     logger.error(
-            "Exception in message processing for: {} :: message: {} {}"
-            , callerName
-            , exception.getMessage(),
-            exception);
+        "Exception in message processing for: {} :: message: {} {}",
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/actors/BaseActor.java:66]: please add context in logger all files

File: group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java [hunk 2]
```diff
@@ -215,12 +215,13 @@ public Response handleMemberAddition(
    * @throws BaseException
    */
   @Override
-  public List<MemberResponse> readGroupMembers(String groupId) throws BaseException {
+  public List<MemberResponse> readGroupMembers(String groupId, Map<String, Object> reqContext)
+      throws BaseException {
 
     List<MemberResponse> members = fetchMembersByGroupIds(Lists.newArrayList(groupId));
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java:221]: add context

File: group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java [hunk 3]
```diff
@@ -87,7 +89,10 @@ public Response searchUserByIds(List<String> userIds) throws BaseException {
   public Response getSystemSettings() throws BaseException {
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java:89]: context need to be passed

File: group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java [hunk 4]
```diff
@@ -105,9 +110,11 @@ public Response getOrganisationDetails(String orgId) throws BaseException {
     Response responseObj = new Response();
     SearchRequest readRequest = new SearchRequest();
     readRequest.getRequest().put(JsonKey.ORGANISATION_ID, orgId);
-
     Map<String, String> requestHeader = new HashMap<>();
-    getUpdatedRequestHeader(requestHeader);
+    Map<String, Object> requestContext = new HashMap<>();
+    requestContext.put(JsonKey.X_REQUEST_ID, UUID.randomUUID().toString());
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java:115]: context should be fro request, not new context

File: sb-utils/src/main/java/org/sunbird/util/JsonKey.java [hunk 5]
```diff
@@ -10,12 +10,14 @@
   String MESSAGE = "message";
   String METHOD = "method";
   String REQUEST_MESSAGE_ID = "msgId";
+  String MESSAGE_ID = "X-msgId";
```
Comment (reshmi-nair) [inline sb-utils/src/main/java/org/sunbird/util/JsonKey.java:13]: Should this 'X' be capital?

File: group-actors/src/main/java/org/sunbird/service/GroupService.java [hunk 6]
```diff
@@ -14,9 +14,10 @@
...
   Map<String, Object> readGroup(String groupId) throws BaseException;
 
-  GroupResponse readGroupWithActivities(String groupId) throws Exception;
+  GroupResponse readGroupWithActivities(String groupId, Map<String, Object> reqContext)
+      throws Exception;
 
-  void readGroupActivities(Map<String, Object> dbResGroup);
+  void readGroupActivities(Map<String, Object> dbResGroup, Map<String, Object> reqContext);
 
   List<GroupResponse> searchGroup(Map<String, Object> searchFilter) throws BaseException;
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/GroupService.java:22]: Context need to be passed

File: group-actors/src/main/java/org/sunbird/service/GroupService.java [hunk 7]
```diff
@@ -14,9 +14,10 @@
 
   Map<String, Object> readGroup(String groupId) throws BaseException;
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/GroupService.java:15]: context need to be passed

File: group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java [hunk 8]
```diff
@@ -96,7 +97,8 @@ public void readGroupActivities(Map<String, Object> dbResGroup) {
    *
    * @param dbResActivities
    */
-  private void addActivityInfoDetails(List<Map<String, Object>> dbResActivities) {
+  private void addActivityInfoDetails(
+      List<Map<String, Object>> dbResActivities, Map<String, Object> reqContext) {
     logger.info("Fetching activityInfo for activity count: {}", dbResActivities.size());
     Map<SearchServiceUtil, Map<String, String>> idClassTypeMap =
         GroupUtil.groupActivityIdsBySearchUtilClass(dbResActivities);
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java:104]: context

File: group-actors/src/main/java/org/sunbird/service/MemberService.java [hunk 9]
```diff
@@ -26,7 +26,8 @@ Response handleMemberAddition(
       List<Map<String, Object>> userGroupsList)
       throws BaseException;
 
-  List<MemberResponse> readGroupMembers(String groupId) throws BaseException;
+  List<MemberResponse> readGroupMembers(String groupId, Map<String, Object> reqContext)
+      throws BaseException;
 
   List<MemberResponse> fetchMembersByGroupId(String groupId) throws BaseException;
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/MemberService.java:32]: pass context

File: group-actors/src/main/java/org/sunbird/service/UserService.java [hunk 10]
```diff
@@ -5,19 +5,29 @@
...
 import org.sunbird.response.Response;
+import org.sunbird.util.JsonKey;
 
 public interface UserService {
 
-  public Response searchUserByIds(List<String> userIds) throws BaseException;
+  public Response searchUserByIds(List<String> userIds, Map<String, Object> context)
+      throws BaseException;
 
   public Response getSystemSettings() throws BaseException;
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/service/UserService.java:15]: pass context

**Review by reshmi-nair — CHANGES_REQUESTED** [review]

File: group-actors/src/main/java/org/sunbird/actors/BaseActor.java (hunk 1 above)
Comment (sknirmalkar89) [reply group-actors/src/main/java/org/sunbird/actors/BaseActor.java:66]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java (hunk 2 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java:221]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java (hunk 3 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java:89]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java (hunk 4 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/UserServiceImpl.java:115]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: sb-utils/src/main/java/org/sunbird/util/JsonKey.java (hunk 5 above)
Comment (reshmi-nair) [reply sb-utils/src/main/java/org/sunbird/util/JsonKey.java:13]: In learner its the same way

File: group-actors/src/main/java/org/sunbird/service/GroupService.java (hunk 6 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/GroupService.java:22]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: group-actors/src/main/java/org/sunbird/service/GroupService.java (hunk 7 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/GroupService.java:15]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

File: group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java (hunk 8 above)
Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java:104]: We use MDC to log traceId in groups, so there is no requirement to pass request context in each method to log traceId

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**